### PR TITLE
Snowflake native path - s3 and GCS

### PR DESCRIPTION
**Pull request by utkarsharma2**
# Description
## What is the current behavior?
Currently, we are doing a non-native transfer in UTO.

closes: https://github.com/astronomer/apache-airflow-provider-transfers/issues/19, https://github.com/astronomer/apache-airflow-provider-transfers/issues/22

## What is the new behavior?
This class is an attempt to introduce native transfer for the snowflake.
Specifically
1. S3 to Snowflake
2. GCS to Snowflake

## Does this introduce a breaking change?
Nope

### Checklist
- [ ] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

File: .github/workflows/ci-uto.yaml [hunk 1]
```diff
@@ -33,7 +33,7 @@ env:
   SNOWFLAKE_ACCOUNT_NAME: ${{secrets.SNOWFLAKE_ACCOUNT_NAME}}
   SNOWFLAKE_SCHEMA: ASTROFLOW_CI
   SNOWFLAKE_DATABASE: SANDBOX
-  SNOWFLAKE_WAREHOUSE: ROBOTS
+  SNOWFLAKE_WAREHOUSE: DEMO
   SNOWFLAKE_HOST: https://gp21411.us-east-1.snowflakecomputing.com
```
Comment (phanikumv) [inline .github/workflows/ci-uto.yaml:37]: this host needs to be in secrets - dont want to expose it

File: src/universal_transfer_operator/data_providers/filesystem/google/cloud/gcs.py [hunk 2]
```diff
@@ -24,6 +24,8 @@ class GCSDataProvider(BaseFilesystemProviders):
     DataProviders interactions with GS Dataset.
     """
 
+    location_type = FileLocation.GS
```
Comment (phanikumv) [inline src/universal_transfer_operator/data_providers/filesystem/google/cloud/gcs.py:27]:
```suggestion
    location_type = FileLocation.GCS
```

File: src/universal_transfer_operator/data_providers/filesystem/google/cloud/gcs.py [hunk 3]
```diff
@@ -24,6 +24,8 @@ class GCSDataProvider(BaseFilesystemProviders):
     DataProviders interactions with GS Dataset.
```
Comment (phanikumv) [inline src/universal_transfer_operator/data_providers/filesystem/google/cloud/gcs.py:24]:
```suggestion
    DataProviders interactions with GCS Dataset.
```

File: tests_integration/test_data_provider/test_databases/test_snowflake.py [hunk 4]
```diff
@@ -156,3 +167,109 @@ def test_write_method(dataset_table_fixture):
...
+    file_dataset = File(
+        path="gs://astro-sdk/workspace/sample.csv", conn_id="google_cloud_default", filetype=FileType.CSV
+    )
+    file_dataprovider = create_dataprovider(dataset=file_dataset)
+    with sample_dag:
+        # [START transfer_from_gcs_to_snowflake_natively]
+        # file_dataset = File(
+        #     path="gs://astro-sdk/workspace/sample.csv", conn_id="google_cloud_default", filetype=FileType.CSV
+        # )
+        # table_dataset = Table(conn_id="snowflake_conn")
```
Comment (phanikumv) [inline tests_integration/test_data_provider/test_databases/test_snowflake.py:234]:
```suggestion
```

Comment (utkarsharma2) [reply tests_integration/test_data_provider/test_databases/test_snowflake.py:234]: @phanikumv We need this in as this code is used in the doc and without this code comment the example won't be meaningful.

File: src/universal_transfer_operator/data_providers/filesystem/google/cloud/gcs.py (hunk 2 above)
Comment (utkarsharma2) [reply src/universal_transfer_operator/data_providers/filesystem/google/cloud/gcs.py:27]: @phanikumv This is aligned with the URL that is constructed - gs://bucket is correct and gcs://bucket will fail, I think it's good to be aligned. WDYT?

File: .github/workflows/ci-uto.yaml (hunk 1 above)
Comment (utkarsharma2) [reply .github/workflows/ci-uto.yaml]: Thanks for pointing this out. Updated this is code.

File: src/universal_transfer_operator/data_providers/filesystem/google/cloud/gcs.py (hunk 2 above)
Comment (phanikumv) [reply src/universal_transfer_operator/data_providers/filesystem/google/cloud/gcs.py:27]: Ok its fine then

**Pull request merged**